sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -11,6 +11,8 @@
 #ifdef WINDOWS
 #include <winsock2.h> // for WSAPoll();
 #endif
+
+#include <algorithm>
```

Comment:
What are we using from this header?

Reply:
std::min()

Reply:
This PR isn't adding calls to `min`. They already exist in master which didn't have this header. Why do we need to include this header now?

https://github.com/Azure/azure-sdk-for-cpp/blob/74020a397a2b2b04bac853953dca7f6f6514d69f/sdk/core/azure-core/src/http/curl/curl.cpp#L598-L610

Its fine to add it explicitly, just wondering. Also, how did you discover this was missing?

Reply:
http.hpp and body_strean.hpp headers both have the include <algorithm>
since curl is including http.hpp, it becomes compilable.

However, if we would move curl component out, it would fail until we add the header.
It needs to be explicitly mentioned.